### DISCUSSION TITLE
Not found

### DISCUSSION THREAD

**Discussion by mateusgalasso** [laravel/nova-issues#4466] [Q&A]
- Laravel Version: 4.9.2
- Nova Version: 9

I have the following relationship between two models: Project hasOne payment. belongsTo Project Payment

When I'm in the Project to create a new Payment I always get the Not Found message.
Laravel Nova is generating the following URL

`http://localhost/nova/resources/payments/new?viaResource=proposals&viaResourceId=35&viaRelationship=payment&relationshipType=hasOne
`
When I adjust to URL

`http://localhost/nova/resources/pagamentos/new?viaResource=propostas&viaResourceId=35`
Payment creation works

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

URL: `/nova/resources/profiles/new?viaResource=users&viaResourceId=5&viaRelationship=profile&relationshipType=hasOne`
![image](https://user-images.githubusercontent.com/172966/174919246-e6a3b0e2-53fb-40dd-b6ec-9220e802eeaa.png)

This taken directly from `laravel/nova-dusk-suite` repository which we used to run our Dusk tests.

**Reply by mateusgalasso** [reply]
But in that case it shouldn't be filled and readonly?
If I understood, Profile -> belongsTo Interests
If you are creating a record like that why yoy have to select? Couldn't come filled by default?

**Reply by crynobone** [reply]
>If I understood, Profile -> belongsTo Interests

No, Profile -> belongsTo User and User is selected to `5` which matches `?viaResource=users&viaResourceId=5&viaRelationship=profile&relationshipType=hasOne`

**Comment by mateusgalasso** [comment]
Even creating a payment via the "Proposta" Resource, Laravel Nova goes through the fields method of the "Proposta" Resource
And in that "Proposal" resource had the following method:

```php
if ($request->editMode == 'create') {
            return [
                BelongsTo::make('Entidade')->searchable()->withoutTrashed(),
                BelongsTo::make('Ação', 'acao', Acao::class)->withoutTrashed(),
            ];
        }
```
The idea here was to show only two fields in create. I know that it has "fieldsForCreate" but I was having problems with it and other packages, so, I did this "if", but as the `$request->editMode == 'create'` came from the resource "Payment", it entered the "if" and did not find the relationships

I changed it to `$request->editMode == 'create' and is_null($request->viaResource)`

now it's working

**Reply by crynobone** [reply]
If you doing something out of suggested approach then it something you have the deal with and solve manually.
